manifests/install.pp

```diff
@@ -69,43 +69,17 @@
     }
   }
 
-  if $group == 'rundeck' and $group_id == '' {
-    ensure_resource('group', 'rundeck', { 'ensure' => 'present' } )
-  }
-  elsif $group == 'rundeck' and $group_id != '' {
-    ensure_resource('group', 'rundeck', { 'ensure' => 'present', 'gid' => $group_id } )
-  }
-  elsif $group != 'rundeck' and $group_id != '' {
-    ensure_resource('group', $group, { 'ensure' => 'present', 'gid' => $group_id } )
-
-    group { 'rundeck':
-      ensure => absent,
-    }
-  }
-  else {
-    ensure_resource('group', $group, { 'ensure' => 'present' } )
+  ensure_resource('group', $group, { 'ensure' => 'present', 'gid' => $group_id } )
```

Comment:
I think it's better that you go all the way and make a breaking change. Something like:
```
# params.pp
$manage_group = false
$manage_user = false
$uid = undef
$gid = undef
# install.pp
if $manage_group {
...
if $manage_user {
...
```

Reply:
I don't disagree with that part (and can probably add those flags), but the bigger question is, should the module _remove_ the rundeck user / group under any circumstances? I'm not totally seeing the utility; it seems like if a package is creating it and the user wants to remove it, maybe it should be handled in site code?

Reply:
> should the module remove the rundeck user / group under any circumstances
Yes, if $manage_user it should be assumed that you're managing it with full consequences, even if ensure => absent doesn't sound like it would be used by anyone

Reply:
What I also need to check is whether the module creates the user / group under normal circumstances, or whether it comes from the package. Otherwise, the module would be totally useless with `$manage_user` and `$manage_group` set to false, right?

Reply:
For sure, on CentOS 7, at least, there's:
```
[root@centos-7-x64 ~]# rpm -q --scripts rundeck
preinstall scriptlet (using /bin/sh):
getent group rundeck >/dev/null || groupadd rundeck
getent passwd rundeck >/dev/null || useradd -d /var/lib/rundeck -m -g rundeck rundeck
```

Reply:
So, even though the user is created by the package on the supported platforms I surveyed, there's still an ordering issue when there are files managed in `install.pp`. We could require `Package['rundeck']` for the `File` resource (kinda hackish), or see if moving some of the config files into `config.pp` works...